_includes/snippets_library/UCL_Myriad_sge/14/gcc_example.snip

```diff
@@ -0,0 +1,172 @@
+```
+{: .bash}
+ 
+```
+------------------------------------- /shared/ucl/apps/modulefiles/applications --------------------------------------
+stata/14 stata/15
+```
+{: .output}
+
+Let's take a closer look at the `matlab` module.
+Matlab is a widely-used piece of software which uses matrix multiplication.
+As we shall see, there are different versions available, and we want to make sure the one we use
+is the correct one for our purposes.
+
+Let's see which versions we have access to.
+
+```
+{{ site.remote.prompt }} module avail matlab
+```
+{: .bash}
+
+```
+------------------------------------- /shared/ucl/apps/modulefiles/applications --------------------------------------
+matlab/full/r2015a/8.5 matlab/full/r2016b/9.1 matlab/full/r2018a/9.4 matlab/full/r2019b/9.7
+matlab/full/r2015b/8.6 matlab/full/r2017a/9.2 matlab/full/r2018b/9.5
+```
+{: .output}
+
+In this case, there are seven different versions.
+How do we load each copy, and which copy is the default?
+
+Sometimes, on some systems, a module might have a `(default)` next to it.
+This indicates that it is the default (i.e. which would be loaded if we type `module load matlab`).
+In this case, we don't see this, so we will have to load matlab and see what we get.
+
+```
+{{ site.remote.prompt }} module load matlab
+```
+{: .bash}
+
+```
+matlab/full/r2019b/9.7(99):ERROR:151: Module 'matlab/full/r2019b/9.7' depends on one of the module(s) 'gcc-libs/4.9.2'
+matlab/full/r2019b/9.7(99):ERROR:102: Tcl command execution failed: prereq gcc-libs
+```
+{: .output}
+
+Here, we see that the default version of Matlab on the system is r2019b/9.7, which 
+in this case is the most recent version. However, you should not assume that the
+default version is necessarily the latest.
+
+As we saw in the earlier example, there are one or more dependencies.
+
+Suppose we decide to load an earlier version of Matlab, e.g. r2017a/9.2.
+
+```
+{{ site.remote.prompt }} module purge
+{{ site.remote.prompt }} module list
+```
+{: .bash}
+
+```
+No Modulefiles Currently Loaded.
+```
+{: .output}
+
+```
+{{ site.remote.prompt }} module load matlab/full/r2017a/9.2
+```
+{: .bash}
+
+```
+matlab/full/r2017a/9.2(96):ERROR:151: Module 'matlab/full/r2017a/9.2' depends on one of the module(s) 'gcc-libs/4.9.2'
+matlab/full/r2017a/9.2(96):ERROR:102: Tcl command execution failed: prereq gcc-libs
+```
+{: .output}
+
+```
+{{ site.remote.prompt }} module load gcc-libs/4.9.2
+{{ site.remote.prompt }} module load matlab/full/r2017a/9.2
+```
+{: .bash}
+
+```
+matlab/full/r2017a/9.2(97):ERROR:151: Module 'matlab/full/r2017a/9.2' depends on one of the module(s) 'xorg-utils/X11R7.7'
+matlab/full/r2017a/9.2(97):ERROR:102: Tcl command execution failed: prereq xorg-utils/X11R7.7
+```
+{: .output}
+
+```
+{{ site.remote.prompt }} module load xorg-utils/X11R7.7
+{{ site.remote.prompt }} module load matlab/full/r2017a/9.2
+```
+{: .bash}
+
+```
+~/.matlab is a symbolic link pointing to /home/cceagil/Scratch/.matlab
```

Comment:
Perhaps replace `cceagil` with the generic `yourUsername`

Reply:
Good catch!